### ISSUE TITLE
Include rebase directly in git-fixup

### ISSUE THREAD

**Issue by zimmski** [keis/git-fixup#33]
git-fixup looks great, imho it lacks just one option that is is to immediately rebase a fixup/squash. If I already know  the commit I want to fixup/squash with I should be able to even save typing the rebase command.

**Comment by keis** [comment]
Sure, that's sounds pretty reasonable even if I myself tend to collect a few fixups before rebasing with `-x make test`. It could be added as a new command line flag (-r|--rebase) and you could alias it if you want that to be the default.

Do you think the interactive editor should show or be bypassed?

**Comment by zimmski** [comment]
In general I test my changes before I do the fixup, since it is usually a bugfix for a recent commit that gets committed. So personally I would like to bypass the interactive editor, but I am sure other people disagree on that. I guess a flag for this interaction would be appropriate as well.

**Comment by keis** [comment]
Right, so something like --interactive maybe. with the default being to suppress the editor

**Comment by quintrino** [comment]
Alternatively, is there a way to have git-fixup return the sha of the commit selected so we can build it into a shell function that first runs regular git-fixup and then you can pass the provided commit to a rebase command?

**Comment by keis** [comment]
So even though the script does have selection now it's not something it's trying to solve. I see two ways I would go about this which is either use tab-completion of the shell, or a menu tool like fzf.

so either create my-rebase.sh and configure zsh to use git-fixup to get the auto-complete data, or have your script do `commit=$(git fixup | fzf)`

I'm not entirely opposed to a simple print flag but I want to make sure it's actually needed. A concrete example of what this rebase script might look like would also be great :)

**Comment by quintrino** [comment]
Sure, so I am currently using this function configured with git-fixup tab completion.
```
compdef gcf=git-fixup

function gcf() {
  git log $@ --oneline -1
  git commit --fixup=$@ && git rebase --autosquash --autostash $@^
}
```

But then I don't get to preview the commit using the fzf fixup menu. I'm just trying to figure out how to combine the two.

**Comment by keis** [comment]
To get the menu you could do something like this, right?
```
function gcf() {
  sha=$(git fixup | fzf)
  git commit --fixup=$sha && git rebase --autosquash --autostash $@^
}
```

**Comment by quintrino** [comment]
I ended up getting 
```
function gcf() {
  sha=$(git fixup | fzf --preview 'git show --color {+1}' | cut -f 1 -d " ")
  git commit --fixup=$sha && git rebase ${sha}^ --autosquash --autostash -i
}
```
to work, although I can't work out why the -i is necessary as without it the message just says `master up to date`
